system/ee/ExpressionEngine/Addons/channel/mod.channel.php

```diff
@@ -3182,6 +3182,17 @@ public function category_archive()
 
         $return_data = '';
 
+        $site_pages = config_item('site_pages');
+
+        foreach (ee()->TMPL->site_ids as $site_id) {
+            if ($site_id != ee()->config->item('site_id')) {
+                $pages = ee()->config->site_pages($site_id);
+                $site_pages[$site_id] = $pages[$site_id];
+            }
+        }
+
+        $site_id = ee()->config->item('site_id');
+
```

Comment:
I am not sure I understand the reason for this chunk of code. I think `$pages = ee()->config->site_pages($site_id);` should be enough?

Reply:
And if I tell you this is a plain copy/paste from parser lib? Let's fix it!

Reply:
That is interesting! I'll need to figure out why this code was build exactly this way - and then hopefully improve it

Reply:
From [here](https://github.com/ExpressionEngine/ExpressionEngine/blob/292951c13a4e6a1605d08b7aac1e86641e39e9b9/system/ee/legacy/libraries/channel_entries_parser/Parser.php#L164), @intoeetive .

Reply:
I've found the reason, @intoeetive . `config_item('site_pages');` returns the pages only for the current site. The loop adds pages from a site or sites declared on a possible `site` parameter.

So, it's right.